compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -313,7 +313,11 @@ private boolean isDistinctSymbol(BSymbol symbol, BSymbol foundSym) {
 
     private boolean hasSameOwner(BSymbol symbol, BSymbol foundSym) {
         // check whether the given symbol owner is same as found symbol's owner
-        if (foundSym.owner == symbol.owner) {
+        if (foundSym.owner == symbol.owner ||
+                //TODO: remove this, once #26284, #30688 are fixed
```

Comment:
Shall we create an issue too?

Reply:
sure

Reply:
created https://github.com/ballerina-platform/ballerina-lang/issues/31433